src/arcsf/eval/metrics.py

```diff
@@ -0,0 +1,98 @@
+import torch
+from rouge_score.rouge_scorer import RougeScorer
+from scipy.stats import ks_2samp
+
+# scorer so it doesn't need to be initialised on every call
+scorer = RougeScorer(["rouge1", "rougeL"], use_stemmer=True)
+
+
+def ks_test(forget: torch.Tensor, retain: torch.Tensor, **kwargs) -> float:
+    """KS test figures 4-6 in the paper
+
+    Args:
+        forget : forget truth ratios
+        retain : retain truth ratios
+
+    Returns:
+        p_value: returns the p_value of the ks_test for use in the forget quality
+    """
+    return ks_2samp(forget, retain, **kwargs).pvalue
+
+
+def eval_accuracy(logits: torch.Tensor, labels: torch.Tensor) -> dict[torch.Tensor]:
+    """Gets the accuracy given logits, this isn't used in the paper.
+
+    Args:
+        logits : output logits
+        labels : targets
+
+    Returns:
+         accuracy : accuracy metric in the form of a dictionary
+    """
+    preds = logits.argmax(dim=-1)
+    accuracy = torch.mean(torch.eq(preds, labels).float())
+    return {"eval_accuracy": accuracy}
+
+
+def conditional_probability(
+    token_normalised_losses: torch.Tensor,
+) -> dict[torch.Tensor]:
```

Comment:
```suggestion
) -> dict[str, torch.Tensor]:
```

but see next comment

Reply:
addressed this